Accessibility cleanup + improvements

ryelle opened the pull request:
This PR cleans up/improves the interactions for screen reader users (mostly) & keyboard-only users.

- Switched the button containers to `nav`, which gives the buttons more context that's missing when you can't see the screen.
- Add the number of events on each day to each cell, so days can be skipped and it's clear when days have no events (it's not just a glitch).
- The filter announces when selected teams change the view, and removing the filter moves the focus back to the select input.

**To test**

Try moving through the calendar with just your keyboard, or by using a screen reader.

tellyworth commented:
When I tab through the calendar view, the currently selected event shows up as a white box in the Twenty Twenty theme. Is that expected?

<img width="318" alt="Screen Shot 2020-03-10 at 10 27 04 am" src="https://user-images.githubusercontent.com/7200686/76265854-c7df6680-62b9-11ea-9c07-320872707cbf.png">

ryelle commented:
I bet it's a gutenberg version difference, I'm running 7.6 from github (might be slightly off from the released version) — do the make sites run the gutenberg plugin, or will we just have what's available in core?

tellyworth commented:
Make sites are running core Gutenberg. I could try enabling the plugin if that's the better solution - the calendar part of our plugin really only needs to run on the main Make site anyway.

ryelle commented:
I'll take another look (my) tomorrow morning without the plugin and see if it's an easy fix  👍

StevenDufresne commented:
> I'll take another look (my) tomorrow morning without the plugin and see if it's an easy fix 👍

I looked into this, 2 issues:

**1. I created the classes based on the idea that the "documentation" team would come back as such. Instead it comes back as "docs".**
 - So change [documentation -> docs ](https://github.com/Automattic/meeting-calendar/blob/9a6cf2c14472ec25afbbce364dc20438163eca45/src/frontend/styles.css#L166) 

**2. Update the default**
 - [add !important to background declaration](https://github.com/Automattic/meeting-calendar/blob/9a6cf2c14472ec25afbbce364dc20438163eca45/src/frontend/styles.css#L127)
- I'm not a big fan of `!important` but will all the cross styling (theme style, gutenberg style... etc) its impossible to predict the selector chains in these 3rd party stylesheets.

ryelle commented:
Updated the defaults, and just added in `.wporg-meeting-calendar__team-docs` as an alternate class for the docs team in case it ever changes (unlikely, but easy to leave in).
